### ISSUE TITLE
Determine best way to include composite actions

### ISSUE THREAD

**Issue by onedr0p** [onedr0p/containers#919]
Currently we use hardcoded repo paths `onedr0p/containers/.github/workflows/simple-checks.yaml@main` in certian steps, it would be great to figure out how to make that more generic for people who are forking this repo and want to build their own containers.

**Comment by br3ndonland** [comment]
The easiest thing to do would be to change the reusable workflow references to [local reusable workflows](https://github.blog/changelog/2022-01-25-github-actions-reusable-workflows-can-be-referenced-locally/). It looks like you're already using local references for the render-readme workflow.

https://github.com/onedr0p/containers/blob/9ee194d926ef4a41a2bce2794f35a925e8b97d64/.github/workflows/release-on-merge.yaml#L37-L42

Happy to get you a PR to update the other ones if it's helpful.

Composite actions are a bit different. You'd have to convert the reusable workflows to `action.yml` files, and they have a bunch of quirks and restrictions (like [this](https://github.com/orgs/community/discussions/41927)). In general I'd recommend sticking with reusable workflows instead of composite actions.

Thanks for maintaining this project. Lots of useful container images.

**Comment by onedr0p** [comment]
I found the previous discussion on it here

https://github.com/onedr0p/containers/pull/918#discussion_r1412777804

**Comment by br3ndonland** [comment]
> I found the previous discussion on it here
>
> [#918 (comment)](https://github.com/onedr0p/containers/pull/918#discussion_r1412777804)

I don't agree with that comment. PRs can _always_ run "with a modified version of the workflow." Any other PR could make the same change shown in https://github.com/onedr0p/containers/pull/918#discussion_r1412777804 and Actions would run that change. I don't think the `uses:` key is really the primary security measure. Other security measures available for PRs:

- [Repos have a setting for preventing GitHub Actions from creating or approving pull requests](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#preventing-github-actions-from-creating-or-approving-pull-requests)
- [`GITHUB_TOKEN` only has read permissions on PRs from forks](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)
- [GitHub secrets don't get passed to workflows that run on PRs from forks](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow)
- If there are further security restrictions needed, secrets can be added to [deployment environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) that require approvals or only run on certain refs

**Comment by bjw-s** [comment]
> Any other PR could make the same change shown in https://github.com/onedr0p/containers/pull/918#discussion_r1412777804 and Actions would run that change.

At the time that we were building this workflow system, I believe we tested this and it would ever only run the workflow contents from the main branch even if the PR specified something else, making this the "easiest" way to do this at the time. Of course insights evolve and we should iterate on / revisit previous results so if there are better ways to do these things today we should definitely take a look at them (when time permits)
